backend/app/middleware/monitoring_middleware.py

```diff
@@ -0,0 +1,117 @@
+"""
+Middleware for automatic request monitoring.
+Captures all requests and sends metrics to the monitoring service.
+"""
+
+import time
+import uuid
+import sys
+import traceback
+import psutil
+from typing import Callable
+from fastapi import Request, Response
+from starlette.middleware.base import BaseHTTPMiddleware
+from starlette.types import ASGIApp
+import logging
+
+from app.monitoring import get_monitoring_service
+
+logger = logging.getLogger("uvicorn")
+
+
+class MonitoringMiddleware(BaseHTTPMiddleware):
+    """Middleware that automatically tracks all HTTP requests."""
+    
+    def __init__(self, app: ASGIApp):
+        super().__init__(app)
+        self.monitoring_service = get_monitoring_service()
+    
+    async def dispatch(self, request: Request, call_next: Callable) -> Response:
+        # Generate unique request ID
+        request_id = str(uuid.uuid4())
+        
+        # Extract request info
+        endpoint = request.url.path
+        method = request.method
+        
+        # Record request start
+        start_time = self.monitoring_service.metrics_collector.record_request_start(
+            request_id=request_id,
+            endpoint=endpoint,
+            method=method
+        )
+        
+        # Initialize metadata
+        metadata = {
+            "user_agent": request.headers.get("user-agent", "unknown"),
+            "client_ip": self._get_client_ip(request),
+        }
+        
+        error = None
+        status_code = 500  # Default to error
+        
+        try:
+            # Process request
+            response = await call_next(request)
+            status_code = response.status_code
+            
+            # Add response headers for tracing
+            response.headers["X-Request-ID"] = request_id
+            
+            return response
+            
+        except Exception as e:
+            error = str(e)
+            
+            # Enhanced error context for crash diagnosis
+            error_context = {
+                "error_type": type(e).__name__,
+                "error_message": str(e),
+                "traceback": traceback.format_exc(),
+            }
+            
+            # Capture memory state at crash time
+            try:
+                process = psutil.Process()
+                mem_info = process.memory_info()
+                error_context.update({
```

Comment:
I would like to see this information (omit the crash_ prefix): memory_rss_md, memory_vms_mb, memory_percent, num_threads logged on *every* request, not just errors.